packages/grafana-ui/src/components/Input/Input.tsx

```diff
@@ -225,6 +237,21 @@ export const Input = React.forwardRef<HTMLInputElement, Props>((props, ref) => {
 
   const theme = useTheme();
   const styles = getInputStyles({ theme, invalid: !!invalid, width });
+  const innerRef = useRef<HTMLInputElement>(null);
+
+  const onNumberInputBlur = useCallback(
+    (e: React.FocusEvent<HTMLInputElement>) => {
+      if (isNaN(parseInt(e.target.value, 10))) {
```

Comment:
It might an idea to use `e.target.validity.valid` here?

Reply:
sure!